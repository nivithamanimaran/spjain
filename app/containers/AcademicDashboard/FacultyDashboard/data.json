{
  "FacultyDashboard": {
    "Assessments": [
      {
        "AssessmentName": "Assessment I",
        "Status": "Resume",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "AssessmentName": "Assessment II",
        "Status": "Publish",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "AssessmentName": "Grade Answer Sheet",
        "Status": "Grade",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "AssessmentName": "Create Question Paper",
        "Status": "Resume",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "AssessmentName": "Assessment V",
        "Status": "Publish",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "AssessmentName": "Assessment VI",
        "Status": "Publish",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "AssessmentName": "Assessment VII",
        "Status": "Publish",
        "Subject": "Principles of Management",
        "Course": "BBA-2022",
        "DueDate": "21/06/2022",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      }
    ],
    "Assignments": [],
    "Resources": [
      {
        "fileName": "chapter1",
        "fileType": "ppt",
        "subject": "Principles of Management",
        "background": "#E3F8FF",
        "color": "#1CAEE4"
      },
      {
        "fileName": "chapter2",
        "fileType": "mp4",
        "subject": "Business Statistics",
        "background": "#FFF4CF",
        "color": "#C29705"
      },
      {
        "fileName": "chapter3",
        "fileType": "mp4",
        "subject": "Financial Accounting",
        "background": "#FFF4CF",
        "color": "#C29705"
      },
      {
        "fileName": "chapter4",
        "fileType": "mp4",
        "subject": "Cost Accounting",
        "background": "#FFF4CF",
        "color": "#C29705"
      },
      {
        "fileName": "chapter5",
        "fileType": "pdf",
        "subject": "Microeconomics 101",
        "background": "#FFEAE4",
        "color": "#E75D3B"
      }
    ],
    "Discussion Forum": []
  },
  "My Subjects": [
    {
      "Subject Name": "Principles of Management",
      "percentage": 75,
      "resource": 15,
      "activities": 5,
      "background": "#FAF0FF",
      "color:": "#C06DE9",
      "NoOfStudents": 90,
      "class": "201-BBA-SE"
    },
    {
      "Subject Name": "Business Statistics",
      "percentage": 66,
      "resource": 5,
      "activities": 20,
      "background": "#FEECEB",
      "color:": "#FF8B85",
      "NoOfStudents": 66,
      "class": "101-BBA-SE"
    },
    {
      "Subject Name": "Financial Accounting",
      "percentage": 59,
      "resource": 0,
      "activities": 5,
      "background": "#FFEBE0",
      "color:": "#FF9C66",
      "NoOfStudents": 74,
      "class": "301-BBA-SE"
    },
    {
      "Subject Name": "Cost Accounting",
      "percentage": 0,
      "resource": 10,
      "activities": 5,
      "background": "#FFFADE",
      "color:": "#BAA226",
      "NoOfStudents": 34,
      "class": "401-BBA-SE"
    },
    {
      "Subject Name": "Microeconomics 101",
      "percentage": 100,
      "resource": 14,
      "activities": 5,
      "background": "#F5FADE",
      "color:": "#94AD23",
      "NoOfStudents": 45,
      "class": "201-BBA"
    },
    {
      "Subject Name": "Operations Research 103",
      "percentage": 50,
      "resource": 19,
      "activities": 5,
      "background": "#FAF0FF",
      "color:": "#C06DE9",
      "NoOfStudents": 67,
      "class": "101-BBA"
    },
    {
      "Subject Name": "Business Communication",
      "percentage": 88,
      "resource": 10,
      "activities": 5,
      "background": "#FEECEB",
      "color:": "#FF8B85",
      "NoOfStudents": 51,
      "class": "301-BBA"
    }
  ]
}
