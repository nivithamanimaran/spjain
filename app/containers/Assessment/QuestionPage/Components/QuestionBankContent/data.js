const data = [
  {
    id: 'category',
    label: 'Category',
    type: 'checkbox',
    open: false,
    value: [],
    options: [],
  },
  {
    id: 'class',
    label: 'Class',
    type: 'checkbox',
    open: false,
    value: [],
    options: [
      {
        label: 'Class III',
        value: 'classIII',
      },
      {
        label: 'Class IV',
        value: 'classIV',
      },
      {
        label: 'Class V',
        value: 'classV',
      },
    ],
  },
  {
    id: 'subject',
    label: 'Subject',
    type: 'radio',
    open: false,
    value: '',
    options: [
      {
        label: 'English',
        value: 'english',
      },
      {
        label: 'Hindi',
        value: 'hindi',
      },
      {
        label: 'Mathematics',
        value: 'matematics',
      },
      {
        label: 'Science',
        value: 'science',
      },
    ],
  },
  {
    id: 'chapter',
    label: 'Chapter',
    type: 'checkbox',
    open: true,
    value: '',
    options: [
      {
        label: 'Stars and the Solar System',
        value: 'stars',
      },
      {
        label: 'Friction',
        value: 'friction',
      },
      {
        label: 'Sound',
        value: 'sound',
      },
      {
        label: 'Nutrition in Plants Class',
        value: 'nutrition',
      },
      {
        label: 'Respiration in Organisms',
        value: 'Respirations',
      },
      {
        label: 'Materials: Metals & Non-Metals',
        value: 'materials_metals_&_nonmetals',
      },
    ],
  },
  {
    id: 'topic',
    label: 'Topic',
    type: 'checkbox',
    open: false,
    value: '',
    options: [
      {
        label: 'Stars and the Solar System',
        value: 'stars',
      },
      {
        label: 'Friction',
        value: 'friction',
      },
      {
        label: 'Sound',
        value: 'sound',
      },
      {
        label: 'Nutrition in Plants Class',
        value: 'nutrition',
      },
      {
        label: 'Respiration in Organisms',
        value: 'Respirations',
      },
      {
        label: 'Materials: Metals & Non-Metals',
        value: 'materials_metals_&_nonmetals',
      },
    ],
  },
];

export default data;
