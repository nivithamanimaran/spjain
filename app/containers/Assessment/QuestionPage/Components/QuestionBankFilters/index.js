import { Grid, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import KenAccordion from '../../../../../components/KenAccordion';
import SearchableCheckList from '../../../../../global_components/SearchableCheckList';
import FilterAccordionLabel from '../../../../../components/UI/FilterAccordionLabel';
import KenButton from '../../../../../global_components/KenButton';

const useStyles = makeStyles(theme => ({
  panelHeader: {
    fontWeight: 600,
    fontSize: '14px',
  },
  clearAllButton: {
    color: theme.palette.KenColors.orange10,
    fontWeight: 600,
    fontSize: '12px',
    textTransform: 'capitalize',
  },
  checklistSectionStyle: {
    maxHeight: '200px',
    overflowY: 'auto',
  },
  accordionSummaryRoot: {
    minHeight: 56,
    '&$expanded': {
      minHeight: 56,
    },
    padding: '0px',
  },
  accordionDetailsRoot: {
    padding: '0px',
  },
  filterItemLabelText: {
    fontSize: '14px',
    fontWeight: '600',
    color: theme.palette.KenColors.neutral900,
  },
}));

const QuestionBankFilters = props => {
  const {
    filterData,
    handleChange,
    handleExpand,
    handleClearFilter,
    getSelectedFiltersCount,
  } = props;
  const classes = useStyles();
  const { t } = useTranslation();

  return (
    <>
      <Grid
        container
        justifyContent="space-between"
        alignItems="center"
        data-testid="questionbank-filters"
        style={{ height: '40px' }}
      >
        <Grid item>
          <Typography className={classes.panelHeader}>
            {t('labels:Filters')}
          </Typography>
        </Grid>
        <Grid item>
          <KenButton onClick={handleClearFilter}>
            <Typography
              className={classes.clearAllButton}
              data-testid="clearall-button"
            >
              {t('labels:Clear_all')}
            </Typography>
          </KenButton>
        </Grid>
      </Grid>
      {filterData.map(item => (
        <KenAccordion
          key={item.id}
          label={
            <FilterAccordionLabel
              label={item.label}
              extraInfoText={getSelectedFiltersCount(item)}
              classOverrides={
                item.open ? { labelText: classes.filterItemLabelText } : {}
              }
            />
          }
          panelId={item.id}
          expanded={item.open}
          onExpand={handleExpand}
          classOverrides={{
            summaryRoot: classes.accordionSummaryRoot,
            detailsRoot: classes.accordionDetailsRoot,
          }}
        >
          {item.type === 'checkbox' && (
            <SearchableCheckList
              isSearchEnabled={true}
              type={item.type}
              value={item.value}
              options={item.options}
              handleChange={handleChange(item.id)}
              checkSectionStyleClass={classes.checklistSectionStyle}
            />
          )}
          {item.type === 'radio' && (
            <SearchableCheckList
              isSearchEnabled={true}
              type={item.type}
              value={item.value}
              name={item.id}
              options={item.options}
              handleChange={handleChange(item.id)}
              checkSectionStyleClass={classes.checklistSectionStyle}
            />
          )}
        </KenAccordion>
      ))}
    </>
  );
};

export default QuestionBankFilters;
