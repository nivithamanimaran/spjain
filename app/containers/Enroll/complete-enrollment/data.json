{
  "Enroll": {
    "Mandatory Courses": [
      {
        "SubjectCode": "BBAECO103",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "SeatsLeft": "1"
      },
      {
        "SubjectCode": "SPJ-BBACOM203",
        "Course": "Macroeconomics",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Finance",
        "SeatsLeft": "3"
      },
      {
        "SubjectCode": "SPJ-BBALIB103",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Marketing",
        "SeatsLeft": "-"
      }
    ],
    "Elective Courses": [
      {
        "SubjectCode": "BBAECO104",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Economics",
        "SeatsLeft": "1"
      },
      {
        "SubjectCode": "SPJ-BBACOM303",
        "Course": "Macroeconomics",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Finance",
        "SeatsLeft": "1"
      },
      {
        "SubjectCode": "SPJ-BBALIB203",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Marketing",
        "SeatsLeft": "9"
      }
    ]
  },
  "Enrolled": {
    "Selected Mandatory Courses": [
      {
        "SubjectCode": "BBAECO103",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "1"
      },
      {
        "SubjectCode": "SPJ-BBACOM203",
        "Course": "Macroeconomics",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "2"
      },
      {
        "SubjectCode": "SPJ-BBALIB103",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "3"
      }
    ],
    "Selected Elective Courses": [
      {
        "SubjectCode": "BBAECO104",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "1"
      },
      {
        "SubjectCode": "SPJ-BBACOM303",
        "Course": "Macroeconomics",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "2"
      },
      {
        "SubjectCode": "SPJ-BBALIB203",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "3"
      }
    ]
  }
}
