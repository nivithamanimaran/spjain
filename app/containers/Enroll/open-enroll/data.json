{
  "Enroll": {
    "Mandatory Courses": [
      {
        "SubjectCode": "BBAECO103",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "SeatsLeft": "1",
        "checked":true
      },
      {
        "SubjectCode": "SPJ-BBACOM203",
        "Course": "Finance",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Hotel Management",
        "SeatsLeft": "3",
        "checked":false
      },
      {
        "SubjectCode": "SPJ-BBALIB103",
        "Course": "Marketing",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Management",
        "SeatsLeft": "12",
        "checked":false
      }
    ],
    "Elective Courses": [
      {
        "SubjectCode": "BBAECO104",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Economics",
        "SeatsLeft": "1",
        "checked":true
      },
      {
        "SubjectCode": "SPJ-BBACOM303",
        "Course": "Hotel Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Finance",
        "SeatsLeft": "1",
        "checked":false
      },
      {
        "SubjectCode": "SPJ-BBALIB203",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Marketing",
        "SeatsLeft": "9",
        "checked":true
      }
    ]
  },
  "Enrolled": {
    "Selected Mandatory Courses": [
      {
        "SubjectCode": "BBAECO103",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "1",
        "checked":true
      },
      {
        "SubjectCode": "SPJ-BBACOM203",
        "Course": "Macroeconomics",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "2",
        "checked":false
      },
      {
        "SubjectCode": "SPJ-BBALIB103",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "3",
        "checked":true
      }
    ],
    "Selected Elective Courses": [
      {
        "SubjectCode": "BBAECO104",
        "Course": "Principles of Management",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "1"
      },
      {
        "SubjectCode": "SPJ-BBACOM303",
        "Course": "Macroeconomics",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "2"
      },
      {
        "SubjectCode": "SPJ-BBALIB203",
        "Course": "Business Communication",
        "Credits": "3.0",
        "Discipline": "Core",
        "PreRequisite": "Micro Economics",
        "No": "3"
      }
    ]
  }
}
