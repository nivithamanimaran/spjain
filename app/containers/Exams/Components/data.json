{
  "Grading Components": [
    {
      "exam":"Quality Management",
      "date":"25-10-2022",
      "time":"10 AM - 1 PM",
      "eligiblity":"Not eligible",
      "seatingPlan":"item.seatingPlan",
      "data": [
        { "label": "Assignment 1", "marks": "20 Marks", "percentage": "50%" },
        { "label": "Assignment 2", "marks": "15 Marks", "percentage": "25%" },
        { "label": "Assignment 3", "marks": "15 Marks", "percentage": "25%" }
      ]
    },
    {
      "exam":"Intro to Marketing",
      "date":"30-10-2022",
      "time":"10 AM - 1 PM",
      "eligiblity":"eligible",
      "seatingPlan":"item.seatingPlan",
      "data": [
        { "label": "Quizz 1", "marks": "20 Marks", "percentage": "50%" },
        { "label": "Quizz 2", "marks": "15 Marks", "percentage": "25%" },
        { "label": "Quizz 3", "marks": "15 Marks", "percentage": "25%" }
      ]
    },
    {
      "exam":"Business Analytics",
      "date":"05-11-2022",
      "time":"10 AM - 1 PM",
      "eligiblity":"eligible",
      "seatingPlan":"item.seatingPlan",
      "data": [
        {
          "label": "Mid-Term Exam 1",
          "marks": "20 Marks",
          "percentage": "50%"
        },
        {
          "label": "Mid-Term Exam 2",
          "marks": "15 Marks",
          "percentage": "25%"
        },
        { "label": "Mid-Term Exam 3", "marks": "15 Marks", "percentage": "25%" }
      ]
    }
  ],
  "Grade Points": [
    {
      "Grade Name": "A+",
      "Range": "86% - 100%",
      "Grade Point": 10
    },
    {
      "Grade Name": "A",
      "Range": "71% - 85%",
      "Grade Point": 9
    },
    {
      "Grade Name": "B+",
      "Range": "56% - 70%",
      "Grade Point": 8
    },
    {
      "Grade Name": "B",
      "Range": "41% - 55%",
      "Grade Point": 7
    },
    {
      "Grade Name": "C",
      "Range": "25% - 41%",
      "Grade Point": 6
    }
  ],
  "Subjects": [
    { "label": "Financial Accounting", "SubjectCode": "FIN 101", "Credits": 3 },
    { "label": "Human Resource Management", "SubjectCode": "HRM 102", "Credits": 3 },
    { "label": "Economics", "SubjectCode": "ECO 103", "Credits": 3 },
    { "label": "Operations Management", "SubjectCode": "OPM 104", "Credits": 3 },
    { "label": "Business Communication Management", "SubjectCode": "BCM 105", "Credits": 3 }
  ]
}
