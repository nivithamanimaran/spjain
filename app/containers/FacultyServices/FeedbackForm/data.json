{
  "Subject": [
    {
      "sno": "1. ",
      "title": "This Subject meets my expectations",
      "rating": 3.5
    },
    {
      "sno": "2. ",
      "title": "The contents of the Subject are very useful",
      "rating": 3
    },
    {
      "sno": "3. ",
      "title": "The Text Book is very helpful",
      "rating": 3
    },
    {
      "sno": "4. ",
      "title": "The Subject is well designed and structured",
      "rating": 3
    },
    {
      "sno": "5. ",
      "title": "The Subject is useful to my career",
      "rating": 3
    },
    {
      "sno": "6. ",
      "title": "Assignments and Examinations were very well designed to test my understanding of the Subject",
      "rating": 3
    },
    {
      "sno": "7. ",
      "title": "Overall, this is an excellent Subject",
      "rating": 3
    }
  ],
  "Faculty": [
    {
      "sno": "1. ",
      "title": "Explains the material clearly and to the point",
      "rating": 3.5
    },
    {
      "sno": "2. ",
      "title": "Is clear with the subject concepts",
      "rating": 3
    },
    {
      "sno": "3. ",
      "title": "Is able to handle the class effectively",
      "rating": 3
    },
    {
      "sno": "4. ",
      "title": "Involves the students in the class",
      "rating": 3
    },
    {
      "sno": "5. ",
      "title": "Accessible to students in and out of the Class",
      "rating": 3
    },
    {
      "sno": "6. ",
      "title": "Relates subject matter with practical examples",
      "rating": 3
    },
    {
      "sno": "7. ",
      "title": "Provides timely feedback on Tests and Assignments",
      "rating": 3
    },
    {
      "sno": "8. ",
      "title": "Is very fair and objective in assessment",
      "rating": 3
    },
    {
      "sno": "9. ",
      "title": "Overall, he/she is an excellent Faculty",
      "rating": 3
    }
  ],
  "Instructions": [
    {
      "title": "Strongly Agree:",
      "rating": 5,
      "class": "strongAgree"
    },
    {
      "title": "Agree:",
      "rating": 4,
      "class": "agree"
    },
    {
      "title": "Neither Agree or Disagree:",
      "rating": 3,
      "class": "neitherAgreeDisagree"
    },
    {
      "title": "Disagree:",
      "rating": 2,
      "class": "disagree"
    },
    {
      "title": "Strongly Disagree:",
      "rating": 1,
      "class": "strongDisagree"
    }
  ]
}
