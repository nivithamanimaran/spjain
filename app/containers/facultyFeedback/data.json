{
    "TableData": [
      {
        "Program": "GMBA",
        "Subject": "Business Analytics",
        "SubjectCode": "BUS101",
        "Feedback": "Submitted"
      },
      {
        "Program": "BBA",
        "Subject": "Introduction to Marketing",
        "SubjectCode": "BUS105",
        "Feedback": "Submitted"
      }, {
        "Program": "BCOM",
        "Subject": "Quality Management",
        "SubjectCode": "MGT301",
        "Feedback": "Pending"
      }
    ]
  }