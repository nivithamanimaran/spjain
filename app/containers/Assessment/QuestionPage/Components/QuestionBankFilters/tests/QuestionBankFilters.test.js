import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import '@testing-library/jest-dom/extend-expect';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../../../utils/themeProvider';
import QuestionBankFilters from '../index';
import { BrowserRouter as Router } from 'react-router-dom';
import { data, data1 } from './data';
afterEach(() => {
  cleanup();
});

const getSelectedFiltersCount = jest.fn();
const handleChange = jest.fn();
const handleExpand = jest.fn();
const handleClearFilter = jest.fn();

describe('Question bank content', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <QuestionBankFilters
            filterData={data}
            getSelectedFiltersCount={getSelectedFiltersCount}
            handleChange={handleChange}
            handleExpand={handleExpand}
            handleClearFilter={handleClearFilter}
          />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankFilters
            filterData={data}
            getSelectedFiltersCount={getSelectedFiltersCount}
            handleChange={handleChange}
            handleExpand={handleExpand}
            handleClearFilter={handleClearFilter}
          />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('questionbank-filters')).toMatchSnapshot();
  });

  it('checkbox should be controlled', () => {
    const { container } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankFilters
            filterData={data}
            getSelectedFiltersCount={getSelectedFiltersCount}
            handleChange={handleChange}
            handleExpand={handleExpand}
            handleClearFilter={handleClearFilter}
          />
        </Router>
      </ThemeProvider>
    );

    let checkbox = container.querySelector("input[name='stars']");
    expect(checkbox).toBeTruthy();
    expect(checkbox).toBeChecked();
  });

  it('checkbox should be controlled 1', () => {
    const { container } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankFilters
            filterData={data}
            getSelectedFiltersCount={getSelectedFiltersCount}
            handleChange={handleChange}
            handleExpand={handleExpand}
            handleClearFilter={handleClearFilter}
          />
        </Router>
      </ThemeProvider>
    );

    let radio = container.querySelectorAll(
      "span[data-testid='radio-group'] input[type=radio]"
    );
    expect(radio[0]).toBeChecked();
  });

  it('Match count in chips based on actions', () => {
    const { getAllByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankFilters
            filterData={data}
            getSelectedFiltersCount={jest.fn(() => 1)}
            handleChange={handleChange}
            handleExpand={handleExpand}
            handleClearFilter={handleClearFilter}
          />
        </Router>
      </ThemeProvider>
    );

    expect(getAllByTestId('selectedfilter-count')[1].textContent).toBe('1');
  });

  it('clearall should work', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <QuestionBankFilters
            filterData={data}
            getSelectedFiltersCount={getSelectedFiltersCount}
            handleChange={handleChange}
            handleExpand={handleExpand}
            handleClearFilter={handleClearFilter}
          />
        </Router>
      </ThemeProvider>
    );

    const clearAll = getByTestId('clearall-button');
    fireEvent.click(clearAll);
    expect(handleClearFilter).toHaveBeenCalledTimes(1);
  });
});
