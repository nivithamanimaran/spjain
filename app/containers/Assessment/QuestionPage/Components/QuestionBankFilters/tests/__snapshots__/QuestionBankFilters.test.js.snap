// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question bank content should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
  data-testid="questionbank-filters"
  style="height: 40px;"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <p
      class="MuiTypography-root makeStyles-panelHeader-1 MuiTypography-body1"
    >
      labels:Filters
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text KenButton__KenButtonDefault-sc-1sg8ful-0 dpKZcv MuiButton-textPrimary"
      data-testid="button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <p
          class="MuiTypography-root makeStyles-clearAllButton-2 MuiTypography-body1"
          data-testid="clearall-button"
        >
          labels:Clear_all
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
